### PR TITLE
Modify solver check and remove shape check for LR and Ridge

### PR DESCRIPTION
**Pull request by Pahandrovich**

### CODE REVIEW

**Comment by Pahandrovich** [comment]
http://intel-ci.intel.com/eb2efcc2-cd8f-f1f5-ad65-8cdcd4b723a1

**Comment by Pahandrovich** [comment]
http://intel-ci.intel.com/eb2fb77f-e756-f1d5-b8d7-8cdcd4b723a1

File: daal4py/sklearn/linear_model/_ridge_0_22.py [hunk 1]
```diff
@@ -101,7 +101,9 @@ def _fit_ridge(self, X, y, sample_weight=None):
             multi_output=True, y_numeric=True)
     self.sample_weight_ = sample_weight
     self.fit_shape_good_for_daal_ = True if X.shape[0] >= X.shape[1] else False
-    if (not self.solver == 'auto' or
+    self.fit_solver_good_for_daal_ = True if self.solver != "sparse_cg" else False
+
+    if (not self.fit_solver_good_for_daal_ or
```
Comment (owerbat) [inline daal4py/sklearn/linear_model/_ridge_0_22.py:106]: We should do the same changes for prediction I suppose

**Comment by Pahandrovich** [comment]
https://intel-ci.intel.com/eb2fc8ef-3c19-f18b-80bb-8cdcd4b723a1?

**Comment by Pahandrovich** [comment]
http://intel-ci.intel.com/eb30b291-da32-f1b0-857f-8cdcd4b723a1

**Pull request closed without merging**